IntegrationTests/Services/AWSEventBridgeIntegrationTests/EventBridgeSigV4ATests.swift

```diff
@@ -90,7 +90,17 @@ class EventBridgeSigV4ATests: XCTestCase {
         let seconds = 20.0
         try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
 
-        endpointId = try await primaryRegionEventBridgeClient.describeEndpoint(input: DescribeEndpointInput(name: endpointName)).endpointId
+        // Retry getting endpointId to ensure it's properly set with exponential backoff
+        var retrievedEndpointId: String?
+        for attempt in 0..<3 {
+            retrievedEndpointId = try await primaryRegionEventBridgeClient.describeEndpoint(input: DescribeEndpointInput(name: endpointName)).endpointId
+            if retrievedEndpointId != nil {
+                break
+            }
+            let delay = pow(2.0, Double(attempt + 1))
+            try await Task.sleep(nanoseconds: UInt64(delay * Double(NSEC_PER_SEC)))
```

Comment:
nit: do math in UInt64 rather than between Doubles
```suggestion
            try await Task.sleep(nanoseconds: UInt64(delay) * NSEC_PER_SEC)
```